### PR TITLE
Neuro evolution project

### PR DESCRIPTION
**Pull request by CMacD12**
:new: :robot:

### What

Neuroevolution Project for Matthew and Cole

### Where

Code in src directory and rst file and necessary images in student-projects directory

### CODE REVIEW

File: src/neuroevolution.py
Comment (jameshughes89) [inline src/neuroevolution.py:1]: Unless you write the full suite of unit tests, I won't be looking at your code and your PR will not be approved.

File: src/visualize.py
Comment (jameshughes89) [inline src/visualize.py:1]: Same as the other. Add unit tests or remove code.

File: site/student-projects/NeuroEvolution/pruned.csv
Comment (jameshughes89) [inline site/student-projects/NeuroEvolution/pruned.csv:1]: This should be in the resources folder under a properly named subdirectory.

File: site/student-projects/NeuroEvolution/config-feedforward
Comment (jameshughes89) [inline site/student-projects/NeuroEvolution/config-feedforward:1]: This may be better served in some other director.

File: site/index.rst [hunk 1]
```diff
@@ -83,7 +83,7 @@ YouTube
     student-lectures/Differential_Evolution/Differential_Evolution
     student-lectures/Search-Space-Analysis/Search_Space_Analysis
     student-lectures/ant-colony/ant-colony-optimization
-    student-lectures/coevolution/coevolution
```
Comment (jameshughes89) [inline site/index.rst:86]: what is this change?

File: site/student-projects/NeuroEvolution/neuroevolution.rst [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+--------------------------------
+
+**Feeding the Network with Data**
+   - The learning process starts with data. This data is fed into the neural network. It typically includes both 'input data' (like images, text, or numbers) and 'expected output data' (the correct answers or classifications).
+
+**Forward Propagation**
+   - Once the input data is fed into the network, it goes through the layers of neurons. Each neuron performs a simple calculation on the data and passes it on to the next layer. This process continues until the output layer produces a result. This result is the network's attempt to give the correct answer based on its current state of knowledge.
+
+**Calculating the Error (Loss)**
+   - The network’s output is then compared to the expected output. The difference between these two is termed as the 'error' or 'loss'. The size of the error indicates how far off the network's prediction is from the actual answer.
```
Comment (jameshughes89) [inline site/student-projects/NeuroEvolution/neuroevolution.rst:41]: Your error vs loss isn't quite correct.

**Review by jameshughes89 — CHANGES_REQUESTED** [review]
The content of the lecture part is fine, but without unit tests, I'm not going to review the code. Either add the unit tests or remove the code and code related things altogether. Change the `literalinclude' to just code blocks with copy/pasted code.

File: site/index.rst (hunk 1 above)
Comment (CMacD12) [reply site/index.rst:86]: That was to fix an error noticed with another student's lecture that meant it wasn't showing in the sidebar and was showing as an error/warning when building on my computer and I noticed it wasn't showing on the site either

File: src/neuroevolution.py
Comment (CMacD12) [reply src/neuroevolution.py:1]: code removed

File: src/visualize.py
Comment (CMacD12) [reply src/visualize.py:1]: code removed

File: site/student-projects/NeuroEvolution/pruned.csv
Comment (CMacD12) [reply site/student-projects/NeuroEvolution/pruned.csv:1]: moved to resources directory

File: site/student-projects/NeuroEvolution/config-feedforward
Comment (CMacD12) [reply site/student-projects/NeuroEvolution/config-feedforward:1]: removed

File: site/student-projects/NeuroEvolution/neuroevolution.rst (hunk 2 above)
Comment (CMacD12) [reply site/student-projects/NeuroEvolution/neuroevolution.rst]: fixed

**Comment by CMacD12** [comment]
All changes should be good

File: resources/neuroevolution-spam-csv/pruned.csv
Comment (jameshughes89) [inline resources/neuroevolution-spam-csv/pruned.csv:1]: If this is not used anymore, remove it

**Review by jameshughes89 — APPROVED** [review]
Looks good. If the csv file is not used anymore, just remove it from the repo.

**Pull request merged**